src/main/main.c

```diff
@@ -234,7 +234,7 @@ void init(void)
     servoMixerInit(masterConfig.customServoMixer);
 #endif
 
-#ifdef SPEKTRUM_BIND
+#if defined(SERIAL_RX) && defined(SPEKTRUM_BIND)
```

Comment:
OK, not living in ideal world ... 

Another possibility is to define target specific differenced in 'target.h' and use `sanitation.h` to define options used in code.
In this case, it would mean that target would define `TARGET_SERIAL_RX_DISABLE` and sanitization code will handle it accordingly. It may blend nicely with makefile target definitions, but depending on `SERIAL_RX` before sanitization would be still impossibile ...